core/src/main/scala/org/http4s/headers/X-B3-Flags.scala

```diff
@@ -0,0 +1,26 @@
+package org.http4s
+package headers
+
+import org.http4s.parser.HttpHeaderParser
+import org.http4s.util.Writer
+import org.http4s.{Header, HeaderKey, ParseResult}
+
+object `X-B3-Flags` extends HeaderKey.Internal[`X-B3-Flags`] with HeaderKey.Singleton {
+  override def parse(s: String): ParseResult[`X-B3-Flags`] =
+    HttpHeaderParser.X_B3_FLAGS(s)
+
+  sealed trait Flag
```

Comment:
It's unlikely to matter here, but if this extends `Product with Serializable`, it makes type inference nicer.